table/block_fetcher.h

```diff
@@ -23,10 +23,11 @@ class BlockFetcher {
   BlockFetcher(RandomAccessFileReader* file,
                FilePrefetchBuffer* prefetch_buffer, const Footer& footer,
                const ReadOptions& read_options, const BlockHandle& handle,
-               BlockContents* contents,
-               const ImmutableCFOptions& ioptions,
+               BlockContents* contents, const ImmutableCFOptions& ioptions,
                bool do_uncompress, const Slice& compression_dict,
-               const PersistentCacheOptions& cache_options)
+               const PersistentCacheOptions& cache_options,
+               Cache* block_cache = nullptr, const Slice* cache_key = nullptr,
```

Comment:
It doesn't handle compressed block cache. Should we handle it in this PR or defer to later?